lake_hylia.logic

```diff
@@ -1,3 +1,98 @@
 area LakeHylia:
-  room BeanstalkArea
-  room Main
+  room BeanstalkArea:
+    node EastArea:
+      door LeftLadder
+      door RightLadder
+    node WestArea:
+      door Ladder
+    node BeanstalkFusion:
+      entrance BeanStalkFusion
+    node BeanStalkFusionExit:
+      exit BeanStalkFusion
+    BeanStalkFusionExit -> WestArea: flag HyliaBeanstalkFusion
+    WestArea -> BeanstalkFusion: flag HyliaBeanstalkFusion
+    node EastArea +: entrance Virtual.Zeffa.FluteAway
+    node WestArea +: entrance Virtual.Zeffa.FluteAway
+  room Main:
+    node Crest:
+      exit Crest
+      flag Virtual.Zeffa.LakeCrest
+    node CrestMinish:
+      door MinishHouse
+      exit CrestHoleFusion
+    Crest -> CrestMinish: item CanTurnMinish
+    CrestMinish -> Crest: item CanGrow
+    Crest -> NorthWest
+    node CrestHoleFusion:
+      entrance CrestHoleFusion
+    CrestMinish -> CrestHoleFusion: flag CrestHoleFusion
+    node MinishWater:
+      exit MinishSouth
+      exit NorthHole
+    node MinishWaterEntrances:
+      entrance South
+      entrance NorthHole
+    MinishWater -> MinishWaterEntrances: item Flippers
+    node EastVineHole:
+      door EastVineHole
+    MinishWater <-> EastVineHole: item Flippers
+    node MinishPortal:
+      door MinishPath
+    MinishPortal <-> MinishWater: item Flippers
+    MinishPortal -> SouthEast: item CanGrow
+    node MinishPortalBonk:
+      flag TreeBonk
+    SouthEast -> MinishPortalBonk: item Boots
+    SouthEast -> MinishPortal: item CanTurnMinish, flag TreeBonk
+    node DigCaveFusionEntrance:
+      entrance DigCaveFusion
+    node DigCaveFusionExit:
+      exit DigCaveFusion
+    Water -> DigCaveFusionEntrance: flag HyliaDigCaveFusion, item MoleMitts, (item Flippers | item Cape)
+    DigCaveFusionExit -> Water: ( flag HyliaDigCaveFusion | item Flippers )
```

Comment:
You can go back into the cave (or flute away) without flippers if the flag is set

Reply:
(Well, flute away with just the flag, going back in still requires mitts)